### ISSUE TITLE
Video seems to lightly stutter every once in a while,

### ISSUE THREAD

**Issue by athanos3d** [RenderHeads/UnityPlugin-AVProVideo#1446]
I am running a video at 48fps (HFR). The stream is a SBS 3D video which looks silky smooth most of the time. However, every once in a while the video will slip a little bit (like a judder issue) that I can usually correct if I pause the video and then un-pause. For the most part the video will looks super smooth for a good 20-30 seconds but the issue always seems to creep back in again.

My question is if there is anything I can do to ensure the video is in complete sync with the frame rate. Like for example, is there a way to pull every frame in Update. 

I am running my system at 192fps, this allows me to visualize in HFR on my monitor: (LEFT FRAME, BFI, RIGHT FRAME, BFI)

Of note is that my system is running at 4 times the framerate, meaning that I want a give frame to be available for 4 consecutive vsyncs so I can show left image, black frame, right image, then finally right frame. This allows me to properly visualize in 3D. My gut feeling is telling me that the read from video is drifting, and that sometimes the frame is available one sub-frame after, which may be causing the jitter. So, if it's possible to anchor video frame reads on a given vsync frame, that might be what I need to do. 

Thank you.

**Your Setup (please complete the following information):**
 - Unity version: 2019.4.131f
 - AVPro Video version (number and edition (trial/core/ultra/enterprise)): core 2.0 (windows only)
 - Operating system version: Win10
 - Device model: 
 - Video specs (resolution, frame-rate, codec, file size): 2840 x 1080 (HD 3D SBS), 48fps, MP4, 1GB

**Comment by athanos3d** [comment]
I just found out about Smooth Video mode, so I turned on this feature and set the Frame Selection to the suggested ElapsedTimeVSyncd, However, the video does not play, it looks like the debug output is saying it's stalled. 

All selections except for None do not work. When selecting None, it defaults to the behavior that I described in my previous message. I've attached a few images that show the debug, one is for none and the other is for ElapsedTimeVSyncd.

I've also included my MediaPlayer setup. Thank you.

![FrameSelection_None](https://user-images.githubusercontent.com/76009139/218283394-2a4206a8-f391-44af-8087-a3c2b99ca9ed.JPG)
![FrameSelection_ElapsedTimeVSync](https://user-images.githubusercontent.com/76009139/218283397-fadf64d2-c52c-4f97-89a3-3aec0af22bb7.jpg)
![MediaPlayerSetup](https://user-images.githubusercontent.com/76009139/218283522-9e138454-5eac-420c-aeea-19c9afdffc42.JPG)

**Comment by athanos3d** [comment]
OK, using WinRT seems to fix the issue, I can run HFR content perfectly now. However, running 24fps content still creates an issue; because I'm running the display at a high refresh rate, it seems as if the 24fps content is doing interpolation  (possibly linear?) between frames. 

Seems like the only way to avoid that is to get Smooth Video mode to work, which I would imagine locks the frame rate read once on the right vsync, So, since I'm running at 192fps, I would imagine that the frame read would happen once every 8 frames (192 / 24). Currently it looks like it's being read every 4 frames, so there's an interpolated frame there which makes everything blurry.

**Comment by Chris-RH** [comment]
Hi @athanos3d,

Smooth Video is only a feature of the Ultra version of AVPro Video.  If you want to test it out, you could try downloading the trial version here: https://github.com/RenderHeads/UnityPlugin-AVProVideo/releases

Have you updated to the latest version of AVPro Video?

What is your hardware setup?

The media info in the images that you have included hint at the video being 60fps, not 24fps.

**Comment by athanos3d** [comment]
Hi Chris, thank you for the message.

I didn't realize that Smooth Video is only for Ultra version, but this is good to know.

A quick question about Smooth Video: if I am running my monitor at a multiple rate of the video (i.e. 192Hz with a 24fps video), will an individual frame be shown 8 times in a row without interpolation, or is there frame interp happening to try to make the experience look smoother? My hopes is that the frame straight up stays as-is until the next frame shows.

I have not updated to the latest APPro, my current version is 2.3.1 (plugin v2.3.0f1-core). 
I am using Unity version 2019.4.13f1

I have not updated because I am a little confused on how the update process works. I'm only able to find trial versions of the latest. Do I install those and because I have a valid license, it will not show as a trial version?

My hardware setup is a Razer laptop with an intel-i7CPU and NVidia GTX1060 w/MaxQ (it's not the latest laptop, but it does the trick :) )

Indeed the image shows 60fps as the video rate, but it exhibits the same behaviour on all videos that I've tried (24, 48 and 60).

Thank you for your time!

**Comment by Chris-RH** [comment]
I would suggest first updating to the latest version of AVPro Video and seeing how it plays.

To update your version of AVPro Video to the latest version:
1. Close your project.
2. Open your project and don't touch anything.
3. Window -> Package Manager
4. In the Package Manager window, click on where it says Unity Registry and scroll down to select My Assets
5. Click on AVPro Video and then on the left there will be the option of first Update
6. Once updated, click on Import and OK

If you do want to test out Smooth Video with the AVPro Video Trial version:
1. Download the trial version
2. Open a new project
3. Drag and drop the trial version into the new project
4. Use Mediaplayer demo scene to test playback

**Comment by athanos3d** [comment]
Hi there, thank you for the message. So the latest plug-in seems to be more robust in Media Foundation mode, but I still see some popping here and there, more the WinRT. I've decided to land on WinRT for what I'm doing, since HFR looks great over long video durations (and so it's better to demo with WinRT).

I did try the Ultra demo version to get Smooth Video, and it definitely works well! I think that it resolves a lot of the stuttering I was seeing. However, I don't think there's any way around the fact that 24fps causes a level of persistence between frames that is very noticeable. No matter what video player I use (VLC for example) or with AVIPro, in 3D or 2D, even when I encode 24fps to 48fps and only double the frames, there is still judder. Using SVPPro tool to do a simple frame overlap interpolation of 24fps to 48fps in real-time seems to help mitigate the issue (which I would assume this is happening on either an application like VUDU when watching 24fps movies, making them look smoother) but brute for frame duplication still looks bad no matter what I'm using.

That said, I can use SVPPro to 'emulate' what a TV does at 24, but at the end 48fps with AI interpolation seems to win on every level for comfort viewing. Since it works on WinRT, I don't think the cost of the upgrade to Ultra makes sense at this time, since I'm only demoing on Windows.

I hope this info is useful, and I truly appreciate your support. Thank you for such an awesome product!!!
